controlplane/kubeadm/controllers/status.go

```diff
@@ -72,6 +72,9 @@ func (r *KubeadmControlPlaneReconciler) updateStatus(ctx context.Context, kcp *c
 	// We are scaling down
 	case replicas > desiredReplicas:
 		conditions.MarkFalse(kcp, controlplanev1.ResizedCondition, controlplanev1.ScalingDownReason, clusterv1.ConditionSeverityWarning, "Scaling down control plane to %d replicas (actual %d)", desiredReplicas, replicas)
+
+		// This means that there was no error in generating the desired number of machine objects
+		conditions.MarkTrue(kcp, controlplanev1.MachinesCreatedCondition)
```

Comment:
Yeah I did, but since this condition encapsulates all of the machines' status, I didn't do that. Eg. if `cloneConfigsAndGenerateMachine ` succeeds for the first machine, and fails somewhere else after that, `MachinesCreatedCondition` will be set to true which will be misleading.  Does that make sense?

Reply:
+1, thanks for the explanation!